### PR TITLE
feat: scripts to install or update uosc with a single line

### PR DESCRIPTION
**Pull request by tomasklaen**
I haven't tested the script for Linux & macOS. If anyone could, it'd be nice :) It should install current latest stable.

I also want to figure out if we can provide a `uosc/update` command that people can just run from their menu and update uosc in the background.

### CODE REVIEW

**Comment by christoph-heinrich** [comment]
Github redirects the download url and curl can't deal with that without adding `-L`, and `~` needs to be expanded manually.
```diff
diff --git a/installers/unix.sh b/installers/unix.sh
index 8d30e88..97872a0 100644
--- a/installers/unix.sh
+++ b/installers/unix.sh
@@ -29,6 +29,8 @@ else
 	abort "This install script works only on linux and macOS."
 fi
 
+data_dir="${data_dir/#\~/$HOME}"
+
 # Remove old and deprecated folders & files
 echo "Deleting old and deprecated uosc files and directories."
 rm -rf "$data_dir/scripts/uosc_shared"
@@ -37,7 +39,7 @@ rm -f "$data_dir/scripts/uosc.lua"
 
 # Install new version
 echo "Downloading: $zip_url"
-curl -o $zip_file $zip_url
+curl -L -o $zip_file $zip_url
 echo "Extracting: $zip_file"
 unzip -od $data_dir $zip_file
 cleanup
@@ -49,7 +51,7 @@ if [ ! -f "$conf_file" ]; then
 	echo "Config not found."
 	mkdir -pv $scriptopts_dir
 	echo "Downloading: $conf_url"
-	curl -o $conf_file $conf_url
+	curl -L -o $conf_file $conf_url
 fi
 
 echo "uosc has been installed."
```

File: installers/unix.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+abort() {
+	cleanup
+	printf "%s\n" "$@" >&2
+	exit 1
+}
+
+# Check OS
+OS="$(uname)"
+if [ "${OS}" == "Linux" ]; then
+	data_dir="${XDG_CONFIG_HOME:-~/.config}/mpv"
```
Comment (christoph-heinrich) [inline installers/unix.sh:25]:
```suggestion
	data_dir="${XDG_CONFIG_HOME:-$HOME/.config}/mpv"
```
`~` doesn't get automatically expanded

File: installers/unix.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+# Remove old and deprecated folders & files
+echo "Deleting old and deprecated uosc files and directories."
+rm -rf "$data_dir/scripts/uosc_shared"
+rm -rf "$data_dir/scripts/uosc"
+rm -f "$data_dir/scripts/uosc.lua"
+
+# Install new version
+echo "Downloading: $zip_url"
+curl -o $zip_file $zip_url
```
Comment (christoph-heinrich) [inline installers/unix.sh:40]:
```suggestion
curl -L -o $zip_file $zip_url
```
curl doesn't deal with the redirect unless `-L` is passed

File: installers/unix.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+cleanup
+
+# Download default config if one doesn't exist yet
+scriptopts_dir="$data_dir/script-opts"
+conf_file="$scriptopts_dir/uosc.conf"
+if [ ! -f "$conf_file" ]; then
+	echo "Config not found."
+	mkdir -pv $scriptopts_dir
+	echo "Downloading: $conf_url"
+	curl -o $conf_file $conf_url
```
Comment (christoph-heinrich) [inline installers/unix.sh:52]: same as above

**Review by christoph-heinrich — CHANGES_REQUESTED** [review]
Can't test on mac, but this is required for linux.

**Comment by tomasklaen** [comment]
Is there no way in mpv to spawn an async subprocess that won't freeze the UI and listen to its stdout and exit events with callbacks?

**Comment by tomasklaen** [comment]
I completely missed that 😐.

**Comment by christoph-heinrich** [comment]
I think it's possible to start a process directly from lua and then treat it's stdin and stdout like a file, but I never tried that myself.

**Comment by tomasklaen** [comment]
> I think it's possible to start a process directly from lua and then treat it's stdin and stdout like a file, but I never tried that myself.

What do you mean by this? Make the process output into a file with `command > file.txt` and read the file periodically?

**Comment by christoph-heinrich** [comment]
> What do you mean by this?

https://www.lua.org/manual/5.1/manual.html#pdf-io.popen

**Comment by tomasklaen** [comment]
Dunno why I always assume we don't have access to stuff like this in here. But I can't find any docs, or snippets on how to use it asynchronously without blocking the thread.

**Pull request merged**